lib/CMakeLists.txt

```diff
@@ -1,6 +1,3 @@
 # SPDX-License-Identifier: Apache-2.0
 
-# The visibility and compilation of libraries may be controlled
-# with a Kconfig setting as follows:
```

Comment:
Sorry I was unclear; I wasn't saying to remove the entire comment, just the commented-out `add_subdirectory_ifdef()` line, since it is reproduced as-is in uncommented form below. I think that continuing to have a comment describing what this line is doing is a good idea, because it is a Zephyr specific CMake extension.

So keeping this comment is good; I didn't mean to ask for it to be removed. That's why I only selected the one line in the review interface, not the entire comment block.

Reply:
I think we should have short comments in the files in this repository which explain what the magic lines they contain are doing, and presume that repository users aren't overly familiar with Zephyr. No objections to adding links to the zephyr docs from here, but brief "don't make me read" style comments in these files is a good thing IMO.

Reply:
Replaced deleted comment with:

>  Conditionally expose library and API headers to enable careful control which source is built.